Shipkov hw2

damatvey opened the pull request:
- Выполнил дз1
- Выполнил дз2

lefasdayan committed:
Update task.md

fixed lesson 2 hw task

damatvey committed:
Выполнил hw2

damatvey committed:
Выполнил hw2
- Добавил классы Cat и Mouse

damatvey committed:
Merge branch 'main' of https://github.com/damatvey/JavaCourse into shipkov-hw2

lefasdayan commented on ru/turing/courses/lesson2/Homework2.java:
```diff
@@ -0,0 +1,79 @@
+
+public class Homework2 {
+    public static void main(String[] args) {
+        //подключаю библиотеку и создаю переменную с текущим годом
```
А как же разбить это все на три отдельных метода?

lefasdayan commented on ru/turing/courses/lesson2/Homework2.java:
```diff
@@ -0,0 +1,79 @@
+
+        //для каждой переменной создаю переменну-буфер, в которой хранится страна:, город: и т.п.
+        String stranaBuffer = scan.next();
+        String strana = scan.next();
```
Лучше называть переменные везде на английском

lefasdayan replied:
!

lefasdayan commented on ru/turing/courses/lesson2/animals/Animal.java:
```diff
@@ -1,4 +1,73 @@
+        this.numberOfPaws = numberOfPaws;
+    }
+
+    protected int numberOfPaws;
```
Лучше перенести наверх к остальным полям

lefasdayan commented on ru/turing/courses/lesson2/animals/Cat.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    public void cheel(String placeOfCheel){
+
```
А реализация?

lefasdayan commented on ru/turing/courses/lesson2/animals/Cat.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    public void play(String playWithLaser){
+
```
Реализация?)

lefasdayan commented on ru/turing/courses/lesson2/animals/Mouse.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    public void eat(String favoriteTypeOfCheese){
+
```
Реализация(

lefasdayan commented on ru/turing/courses/lesson2/animals/Mouse.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    public void sleep(int hoursOfsleep){
+
```
РеАлИзАцИя

lefasdayan commented on ru/turing/courses/lesson2/animals/Mouse.java:
```diff
@@ -0,0 +1,39 @@
+
+    }
+    public void krodetsya(){
+        System.out.println("Мыш......(кродеться)...");
```
хехе

Diagorn replied:
Хехе

lefasdayan commented on ru/turing/courses/lesson2/Homework2.java:
```diff
@@ -0,0 +1,79 @@
+        System.out.println("ул." + ulica);
+        System.out.println("д." + dom);
+        System.out.println("кв." + kvartira);
+
```
Было бы неплохо добавить в Main реализацию животных и их методов

damatvey committed:
Внёс изменения в hw2
- Вынес составление адреса в краткой форме в отдельный метод
- Добавил реализацию методам
- Добавил в Main реализацию животных и их методов

lefasdayan commented on ru/turing/courses/lesson2/Homework2.java:
```diff
@@ -0,0 +1,100 @@
+
+        String kvartiraBuffer = scan.next();
+        String kvartira = scan.next();
+        kvartira = kvartira.replace(String.valueOf(charToRemove), "");
```
Внутреннюю логику методов надо засунуть в сами тела методов. В мэйне -- только вызывать методы

lefasdayan commented on ru/turing/courses/lesson2/animals/Animal.java:
```diff
@@ -1,4 +1,72 @@
 package ru.turing.courses.lesson2.animals;
 
 public class Animal {
+    protected String klichka;
```
```suggestion
    protected String nickname;
```
Называть по возможности словами на английском языке!

lefasdayan replied:
!

damatvey committed:
Доделал hw2
- Внес логику методо в тела методов
- Не стал переименовывать имена переменных, на будущее учту

lefasdayan commented on ru/turing/courses/lesson2/Homework2.java:
```diff
@@ -0,0 +1,105 @@
+        cat.cheel();
+
+        //подключаю библиотеку и создаю переменную с текущим годом
+        Calendar calendar = new GregorianCalendar();
+        int currentYear = calendar.get(Calendar.YEAR);
```
Это всё можно засунуть внутрь метода

lefasdayan commented on ru/turing/courses/lesson2/Homework2.java:
```diff
@@ -0,0 +1,105 @@
+        strana = strana.replace(String.valueOf(charToRemove), "");
+
+        //Повторяю те же действия для каждой составляющей адресса
+        String gorodBuffer = scan.next();
```
```suggestion
        String cityBuffer = scan.next();
```
И дальше тоже исправь

lefasdayan commented on ru/turing/courses/lesson2/animals/Animal.java:
```diff
@@ -1,4 +1,72 @@
+    protected int age;
+    protected int numberOfPaws;
+
+    public String getKlichka() {
```
Соответственно с названием полей переименовать геттеры/сеттеры и конструкторы в наследниках

lefasdayan commented on ru/turing/courses/lesson2/animals/Cat.java:
```diff
@@ -0,0 +1,43 @@
+        this.favoriteTypeOfFeed = favoriteTypeOfFeed;
+    }
+
+    public void play(String playWithLaser) {
```
```suggestion
    @Override
    public void play(String playWithLaser) {
```
И эту же аннотацию над всеми переопределенными от родителя методами

damatvey committed:
Доделал hw2
- переименовал переменный
- пере создал геттеры-сеттеры, конструкторы
- сделал аннотации

lefasdayan commented:
Еееее бойййййй 🥇
